src/atomate2/vasp/flows/defect.py

```diff
@@ -22,28 +30,127 @@
+# Default settings                                                            ##
+################################################################################
 
 DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
 DEFECT_INCAR_SETTINGS = {
     "ISMEAR": 0,
-    "LWAVE": True,
     "SIGMA": 0.05,
     "KSPACING": None,
     "ENCUT": 500,
 }
-DEFECT_KPOINT_SETTINGS = {"reciprocal_density": 64}
+DEFECT_KPOINT_SETTINGS = Kpoints(
+    comment="special k-point",
+    num_kpts=1,
+    style=Kpoints_supported_modes.Reciprocal,
+    kpts=((0.25, 0.25, 0.25),),
+    kpts_shift=(0, 0, 0),
+    kpts_weights=[1],
+)
 
-DEFECT_RELAX_GENERATOR = AtomicRelaxSetGenerator(
+DEFECT_RELAX_GENERATOR: AtomicRelaxSetGenerator = AtomicRelaxSetGenerator(
     use_structure_charge=True,
     user_incar_settings=DEFECT_INCAR_SETTINGS,
     user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
 )
-DEFECT_STATIC_GENERATOR = StaticSetGenerator(
+
+DEFECT_STATIC_GENERATOR: StaticSetGenerator = StaticSetGenerator(
     user_incar_settings=DEFECT_INCAR_SETTINGS,
     user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
 )
 
 
+################################################################################
+# Formation Energy                                                            ##
+################################################################################
+
+
+@dataclass
+class FormationEnergyMaker(Maker):
+    """Class to generate VASP input sets for the calculation of the formation energy
+    diagram.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow created by this maker.
+    relax_maker: .BaseVaspMaker or None
+        A maker to perform a atomic-position-only relaxation on the defect charge
+        states.
+        If None, the defaults will be used.
+    """
+
+    name: str = "formation energy"
+
+    relax_maker: BaseVaspMaker = field(
+        default_factory=lambda: RelaxMaker(
+            input_set_generator=DEFECT_RELAX_GENERATOR,
+            task_document_kwargs={"store_volumetric_data": ["locpot"]},
+        )
+    )
+
+    def __post_init__(self):
+        """Post-initialization."""
+        self.relax_maker = self.relax_maker.update_kwargs(
+            update={
+                "_set": {
+                    "input_set_generator->user_incar_settings->LVHAR": True,
+                    "input_set_generator->use_structure_charge": True,
+                }
+            },
+            class_filter=RelaxMaker,
+            dict_mod=True,
+        )
+        self.relax_maker = self.relax_maker.update_kwargs(
+            update={
+                "_add_to_set": {"task_document_kwargs->store_volumetric_data": "locpot"}
+            },
+            class_filter=RelaxMaker,
+            dict_mod=True,
+        )
+
+    def make(
+        self,
+        defect_gen: DefectGenerator,
+        dielectric: float | NDArray,
+        bulk_sc_dir: str | Path | None = None,
+        sc_mat: NDArray | None = None,
```

Comment:
Can you make these names more descriptive. E.g.:

- `defect_gen` -> `defect_generator`
- `bulk_sc_dir` -> `bulk_supercell_dir`
- `sc_mat` -> `supercell_matrix`

I know these are a bit verbose but that is just the python way.